pandas/core/arrays/string_.py

```diff
@@ -197,7 +197,10 @@ def __arrow_array__(self, type=None):
 
         if type is None:
             type = pa.string()
-        return pa.array(self._ndarray, type=type, from_pandas=True)
+
+        values = self._ndarray.copy()
```

Comment:
does this copy become unnecessary at some point in the future when pyarrow becomes aware of pd.NA?

Reply:
> does this copy become unnecessary at some point in the future when pyarrow becomes aware of pd.NA?

Yep. For now we need to turn the NAs into Nones (or nans) so that the conversion code from pyarrow works.